js/ui/form/ui.form.layout_manager.js

```diff
@@ -436,12 +436,18 @@ var LayoutManager = Widget.inherit({
     },
 
     _getColCount: function() {
-        var colCount = this.option("colCount"),
+        var screenFactor, currentColCount,
+            colCount = this.option("colCount"),
             colCountByScreen = this.option("colCountByScreen");
 
         if(colCountByScreen) {
-            var screenFactor = windowUtils.hasWindow() ? windowUtils.getCurrentScreenFactor(this.option("screenByWidth")) : "lg",
-                currentColCount = colCountByScreen[screenFactor];
+            var formTargetScreenFactor = this.option("form").getTargetScreenFactor();
+            if(formTargetScreenFactor) {
+                screenFactor = formTargetScreenFactor;
+            } else {
+                screenFactor = windowUtils.hasWindow() ? windowUtils.getCurrentScreenFactor(this.option("screenByWidth")) : "lg";
+            }
```

Comment:
I suggest:
`let screenFactor = this.option("form").getTargetScreenFactor();`
Use `||`
`screenFactor = screenFactor || windowUtils.hasWindow() ? windowUtils.getCurrentScreenFactor(this.option("screenByWidth")) : "lg";`
Or
```
if(!screenFactor) {
	screenFactor = windowUtils.hasWindow() ? windowUtils.getCurrentScreenFactor(this.option("screenByWidth")) : "lg";
}
```